### PR TITLE
Add djinni-support-lib 1.0.0

### PR DESCRIPTION
**Pull request by jothepro**
Specify library name and version:  **djinni-support-lib/1.0.0**

major version update of djinni-support-lib to v1.0.0

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by a4z — DISMISSED** [review]

**Review by prince-chrismc — DISMISSED** [review]

File: recipes/djinni-support-lib/1.x.x/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.4)
+project(conan_cmake_wrapper)
+
+include(conanbuildinfo.cmake)
+conan_basic_setup()
+
+if(WIN32 AND BUILD_SHARED_LIBS)
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif()
```
Comment (uilianries) [inline recipes/djinni-support-lib/1.x.x/CMakeLists.txt:9]:
```suggestion
if(WIN32 AND BUILD_SHARED_LIBS)
    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
endif()
```

As you are the author, you could fix it directly on upstream for a next release.

File: recipes/djinni-support-lib/1.x.x/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                        "target": "auto",
+                        "system_java": False,
+                       }
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+
+    _cmake = None
+
+    @property
+    def objc_support(self):
```
Comment (uilianries) [inline recipes/djinni-support-lib/1.x.x/conanfile.py:30]:
```suggestion
    def _objc_support(self):
```

File: recipes/djinni-support-lib/1.x.x/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        if self.options.target == "auto":
+            return tools.is_apple_os(self.settings.os)
+        else:
+            return self.options.target == "objc"
+
+    @property
+    def jni_support(self):
```
Comment (uilianries) [inline recipes/djinni-support-lib/1.x.x/conanfile.py:37]:
```suggestion
    def _jni_support(self):
```

File: recipes/djinni-support-lib/1.x.x/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        if self.options.target == "auto":
+            return self.settings.os == "Android"
+        else:
+            return self.options.target == "jni"
+
+    @property
+    def python_support(self):
```
Comment (uilianries) [inline recipes/djinni-support-lib/1.x.x/conanfile.py:44]:
```suggestion
    def _python_support(self):
```

File: recipes/djinni-support-lib/1.x.x/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        return self.options.target == "python"
+
+    @property
+    def cppcli_support(self):
```
Comment (uilianries) [inline recipes/djinni-support-lib/1.x.x/conanfile.py:48]:
```suggestion
    def _cppcli_support(self):
```

File: recipes/djinni-support-lib/1.x.x/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+            if self.cppcli_support:
+                raise ConanInvalidConfiguration("C++/CLI does not support building as shared library")
+        if self.settings.get_safe("compiler.cppstd"):
+            tools.check_min_cppstd(self, "17")
+        try:
+            minimum_required_compiler_version = self._supported_compilers[str(self.settings.compiler)]
+            if tools.Version(self.settings.compiler.version) < minimum_required_compiler_version:
+                raise ConanInvalidConfiguration("This package requires c++17 support. The current compiler does not support it.")
+        except KeyError:
+            self.output.warn("This recipe has no support for the current compiler. Please consider adding it.")
```
Comment (uilianries) [inline recipes/djinni-support-lib/1.x.x/conanfile.py:97]:
```suggestion
    def configure(self):
        if self.options.shared:
            del self.options.fPIC

    def validate(self):
        if not (self.objc_support or self.jni_support or self.python_support or self.cppcli_support):
            raise ConanInvalidConfiguration("Target language could not be determined automatically. Set target explicitly to one of 'jni', 'objc', 'python' or 'cppcli'")
        if self.cppcli_support and (self.settings.compiler.runtime == "MT" or self.settings.compiler.runtime == "MTd"):
            raise ConanInvalidConfiguration("'/clr' and '/MT' command-line options are incompatible")
        if self.options.shared:
            if self.cppcli_support:
                raise ConanInvalidConfiguration("C++/CLI does not support building as shared library")
        if self.settings.get_safe("compiler.cppstd"):
            tools.check_min_cppstd(self, "17")
        try:
            minimum_required_compiler_version = self._supported_compilers[str(self.settings.compiler)]
            if tools.Version(self.settings.compiler.version) < minimum_required_compiler_version:
                raise ConanInvalidConfiguration("This package requires c++17 support. The current compiler does not support it.")
        except KeyError:
            self.output.warn("This recipe has no support for the current compiler. Please consider adding it.")
```

File: recipes/djinni-support-lib/1.x.x/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
```
Comment (uilianries) [inline recipes/djinni-support-lib/1.x.x/conanfile.py:101]:
```suggestion
        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```

**Review by uilianries — CHANGES_REQUESTED** [review]
Thanks for your contribution.

- Rename public properties, they should be protected. It avoid name collision in case Conan adds same method/property in the future
- Use validate() instead of configure() for compiler and other compatibilities.

File: recipes/djinni-support-lib/1.x.x/conanfile.py (hunk 2 above)
Comment (jothepro) [reply recipes/djinni-support-lib/1.x.x/conanfile.py:30]: Should I update the [old conanfile](https://github.com/conan-io/conan-center-index/blob/master/recipes/djinni-support-lib/all/conanfile.py#L29) as well, while we're at it? Or should I keep it unchanged or fix it in a separate PR?

**Review by a4z — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**